provider/rediscloud_essentials_database_test.go

```diff
@@ -0,0 +1,182 @@
+package provider
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"regexp"
+	"testing"
+)
+
+func TestAccRedisCloudEssentialsDatabase_BasicCRUDI(t *testing.T) {
+
+	subscriptionName := acctest.RandomWithPrefix(testResourcePrefix)
+	databaseName := subscriptionName + "-db"
+	databaseNameUpdated := databaseName + "-updated"
+
+	resourceName := "rediscloud_essentials_database.example"
+	datasourceName := "data.rediscloud_essentials_database.example"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: providerFactories,
+		CheckDestroy:      testAccCheckEssentialsSubscriptionDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: fmt.Sprintf(testAccResourceRedisCloudEssentialsDatabaseBasic, subscriptionName, databaseName),
+				Check: resource.ComposeTestCheckFunc(
+					// Test the resource
+					resource.TestMatchResourceAttr(resourceName, "id", regexp.MustCompile("^\\d+/\\d+$")),
+					resource.TestCheckResourceAttrSet(resourceName, "subscription_id"),
+					resource.TestCheckResourceAttrSet(resourceName, "db_id"),
+					resource.TestCheckResourceAttr(resourceName, "name", databaseName),
+					resource.TestCheckResourceAttr(resourceName, "protocol", "stack"),
+					resource.TestCheckResourceAttr(resourceName, "cloud_provider", "AWS"),
+					resource.TestCheckResourceAttr(resourceName, "region", "eu-west-1"),
+					resource.TestCheckResourceAttrSet(resourceName, "redis_version_compliance"),
+					resource.TestCheckResourceAttr(resourceName, "resp_version", "resp2"),
+					resource.TestCheckResourceAttr(resourceName, "data_persistence", "none"),
+					resource.TestCheckResourceAttr(resourceName, "data_eviction", "volatile-lru"),
+					resource.TestCheckResourceAttr(resourceName, "replication", "false"),
+					resource.TestCheckResourceAttrSet(resourceName, "public_endpoint"),
+					resource.TestCheckResourceAttr(resourceName, "private_endpoint", ""),
+
+					// Can't get this to work
```

Comment:
should this be working? If it's ok for it not to maybe the test should be removed

Reply:
Ah yes, I just couldn't create an example that worked in real life but the attribute is communicated as expected. Redis are going to test a release candidate so they'll confirm whether or not everything is OK here.